sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsCheck.java

```diff
@@ -0,0 +1,160 @@
+ * <ul>
+ * <li> System.exit() should not be called.</li>
+ * <li> Assert.assertTrue() has a 1 arg variant that does not provide a helpful message on failure.
+ * <li> Thread.sleep() can be prohibited.
+ * </ul>
+ * E.g.:
+ * <p>
+ * {@code
+ * assertTrue(condition);
+ * }
+ * </p>
+ *
+ * @author <a href="mailto:raghavgautam@gmail.com">Raghav Kumar Gautam</a>
+ */
+public class ForbidCertainMethodsCheck extends AbstractCheck {
+
+    /** Key is pointing to the warning message text in "messages.properties" file. */
+    public static final String MSG_KEY_WITH_ARG = "forbid.certain.methods";
+    /** Key is pointing to the warning message text in "messages.properties" file. */
+    public static final String MSG_KEY_WITHOUT_ARG = "forbid.certain.methods.noarg";
+    /** Simple name of this class. */
+    private static final String CLASS_NAME = ForbidCertainMethodsCheck.class.getSimpleName();
+
+    /** Name of the method. */
+    private Pattern methodName;
+
+    /** Regex/String for number of arguments. */
+    private Pattern argumentCount;
+
+    /**
+     * Set method name regex for the forbidden method.
+     * @param methodName regex for the method name
+     * @throws CheckstyleException when methodName is not a valid regex
+     */
+    public void setMethodName(Pattern methodName) throws CheckstyleException {
+        this.methodName = methodName;
+    }
+
+    /**
+     * Set arg count regex for the forbidden method, can be null.
+     * @param argumentCount regex for argument count
+     * @throws CheckstyleException when argumentCount is not a valid regex
+     */
+    public void setArgumentCount(Pattern argumentCount) throws CheckstyleException {
+        this.argumentCount = argumentCount;
+    }
+
+    @Override
+    public void init() {
+        if (methodName == null) {
+            throw new IllegalStateException(String.format(
+                "empty value for methodName (%s) is not allowed in %s",
+                methodName, CLASS_NAME));
+        }
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.METHOD_CALL,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.METHOD_CALL:
+                final DetailAST dot = ast.getFirstChild();
+                // method that looks like: method()
+                final String methodNameInCode;
+                final int numArgsInCode;
+                if (dot.getType() == TokenTypes.IDENT) {
+                    methodNameInCode = dot.getText();
+                    numArgsInCode = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                    checkForbiddenMethod(ast, methodNameInCode, numArgsInCode);
```

Comment:
this call is already after the if/else, so this duplication can be removed.

Reply:
Done.